tools/shared/src/main/scala/org/scalajs/core/tools/linker/frontend/optimizer/GenIncOptimizer.scala

```diff
@@ -1027,31 +1029,37 @@ object GenIncOptimizer {
 
   private[optimizer] trait AbsCollOps {
     type Map[K, V] <: mutable.Map[K, V]
-    type ParMap[K, V] <: GenMap[K, V]
-    type AccMap[K, V]
-    type ParIterable[V] <: GenIterableLike[V, ParIterable[V]]
-    type Addable[V]
+    type ParMap[K, V] <: AnyRef
+    type AccMap[K, V] <: AnyRef
+    type ParIterable[V] <: AnyRef
+    type Addable[V] <: AnyRef
 
     def emptyAccMap[K, V]: AccMap[K, V]
     def emptyMap[K, V]: Map[K, V]
     def emptyParMap[K, V]: ParMap[K, V]
     def emptyParIterable[V]: ParIterable[V]
 
     // Operations on ParMap
+    def parMapIsEmpty[K, V](map: ParMap[K, V]): Boolean
```

Comment:
Simply name this `isEmpty`? `parMap` seems superfluous (I understand that overloading is a problem with types that end up being erased, but the whole thing is already terribly inconsistent anyways :S).